bindings/python/pymongoarrow/api.py

```diff
@@ -190,16 +192,20 @@ def _arrow_to_numpy(arrow_table, schema):
     See https://arrow.apache.org/docs/python/numpy.html for details.
     """
     container = {}
-    for fname in schema:
-        dtype = get_numpy_type(schema.typemap[fname])
-        if dtype == np.str_:
-            container[fname] = arrow_table[fname].to_pandas().to_numpy(dtype=dtype)
-        else:
-            container[fname] = arrow_table[fname].to_numpy()
+    if schema:
+        for fname in schema:
+            dtype = get_numpy_type(schema.typemap[fname])
+            if dtype == np.str_:
+                container[fname] = arrow_table[fname].to_pandas().to_numpy(dtype=dtype)
+            else:
+                container[fname] = arrow_table[fname].to_numpy()
+    else:
+        for col in arrow_table.column_names:
+            container[col] = arrow_table[col].to_numpy()
```

Comment:
Looks like we're loosing the `if dtype == np.str_:` logic. Is that important? Can we add a test case if so?